src/uiScreens.c

```diff
@@ -461,17 +442,115 @@ void ui_displayPoolOwnerScreen(
 			ownerDescription[descLen] = '\0';
 		}
 
-		descLen += humanReadableAddress(
-		                   rewardAddress, SIZEOF(rewardAddress),
-		                   ownerDescription + descLen, SIZEOF(ownerDescription) - descLen
-		           );
+		{
+			uint8_t rewardAddress[REWARD_ACCOUNT_SIZE];
+
+			switch (owner->keyReferenceType) {
+			case KEY_REFERENCE_PATH: {
+				constructRewardAddressFromKeyPath(
+				        &owner->path, networkId, rewardAddress, SIZEOF(rewardAddress)
+				);
+				break;
+			}
+			case KEY_REFERENCE_HASH: {
+				constructRewardAddressFromKeyHash(
+				        networkId,
+				        owner->keyHash, SIZEOF(owner->keyHash),
+				        rewardAddress, SIZEOF(rewardAddress)
+				);
+				break;
+			}
+			default:
+				ASSERT(false);
+			}
+
+			descLen += humanReadableAddress(
```

Comment:
when the reward address is passed as a path, we should show that to the operator to give them a reasonable guarantee that they control the reward address and see what path it belongs to